### PR TITLE
Fix compiler warnings in compilation units (-Wsign-conversion)

### PR DESCRIPTION
**Pull request by fhuberts**
re-opened from #836

### CODE REVIEW

File: src/oatpp/parser/json/mapping/Deserializer.cpp [hunk 1]
```diff
@@ -494,7 +494,7 @@ oatpp::Void Deserializer::deserializeObject(Deserializer* deserializer, parser::
 }
 
 oatpp::Void Deserializer::deserialize(parser::Caret& caret, const Type* const type) {
-  auto id = type->classId.id;
+  v_uint32 id = static_cast<v_uint32>(type->classId.id);
```
Comment (lganzzzo) [inline src/oatpp/parser/json/mapping/Deserializer.cpp:497]: We can use `auto` here

File: src/oatpp/parser/json/mapping/Serializer.cpp [hunk 2]
```diff
@@ -259,7 +259,7 @@ void Serializer::serializeObject(Serializer* serializer,
 void Serializer::serialize(data::stream::ConsistentOutputStream* stream,
                            const oatpp::Void& polymorph)
 {
-  auto id = polymorph.getValueType()->classId.id;
+  v_uint32 id = static_cast<v_uint32>(polymorph.getValueType()->classId.id);
```
Comment (lganzzzo) [inline src/oatpp/parser/json/mapping/Serializer.cpp:262]: auto here:

```cpp
auto id = static_cast<v_uint32>(polymorph.getValueType()->classId.id);
```

**Review by lganzzzo — COMMENTED** [review]
Hello @fhuberts ,

Looks good to me.
Please fix two minor typos

Also, I have a minor concern about portability of `socklen_t `.
It might be a good idea if we add oatpp specific type for that (/core/IODefinitions.hpp):

Something like:

```cpp
#ifdef _WIN32
    typedef int v_sock_size; 
#else
    typedef socklen_t v_sock_size;
#endif
```

**Comment by fhuberts** [comment]
Done

File: src/oatpp/parser/json/mapping/Deserializer.cpp (hunk 1 above)
Comment (fhuberts) [reply src/oatpp/parser/json/mapping/Deserializer.cpp]: done

File: src/oatpp/parser/json/mapping/Serializer.cpp (hunk 2 above)
Comment (fhuberts) [reply src/oatpp/parser/json/mapping/Serializer.cpp]: done

**Review by lganzzzo — APPROVED** [review]

**Comment by lganzzzo** [comment]
Merged!

**Pull request merged**